pkg/scheduler/algorithm/priorities/image_locality.go

```diff
@@ -69,12 +71,21 @@ func calculateScoreFromSize(sumSize int64) int {
 }
 
 // totalImageSize returns the total image size of all the containers that are already on the node.
+// TODO: cover the corner cases of missed matches, e.g,
+// 1. Using Docker as runtime and docker.io/library/test:tag in pod spec, but only test:tag will present in node status
+// 2. Using the implicit registry, i.e., test:tag or library/test:tag in pod spec but only docker.io/library/test:tag
+// in node status; note that if users consistently use one registry format, this should not happen.
 func totalImageSize(nodeInfo *schedulercache.NodeInfo, containers []v1.Container) int64 {
 	var total int64
 
 	imageSizes := nodeInfo.ImageSizes()
 	for _, container := range containers {
-		if size, ok := imageSizes[container.Image]; ok {
+		// apply the default image tag to ensure matching CRI compliant image names
+		name := container.Image
+		if strings.LastIndex(name, ":") <= strings.LastIndex(name, "/") {
+			name = name + ":" + parsers.DefaultImageTag
+		}
+		if size, ok := imageSizes[name]; ok {
```

Comment:
The change looks good! Thanks.

Could you please extract this logic into a separate function, with unit test, and move the `TODO: cover the corner ...` to it? It would be much clearer to tell out what we support and missed by test cases.

A example is `TestApplyDefaultImageTag` in `image_manager_tests.go`

Reply:
Done. Thanks!